src/python/TaskWorker/MasterWorker.py

```diff
@@ -524,16 +560,29 @@ def algorithm(self):
         self.restartQueuedTasks()
         self.logger.debug("Master Worker Starting Main Cycle.")
         while not self.STOP:
-            selection_limit = self.config.TaskWorker.task_scheduling_limit
-            if not self._selectWork(limit=selection_limit):
-                self.logger.warning("No tasks selected.")
-            else:
-                self.logger.info("Work selected successfully.")
+            is_canary = self.config.TaskWorker.is_canary
+            canary_name = self.config.TaskWorker.canary_name
             limit = self.slaves.queueableTasks()
-            if not self._lockWork(limit=limit, getstatus='NEW', setstatus='HOLDING'):
-                time.sleep(self.config.TaskWorker.polling)
-                continue
 
+            # _selectWork and _lockWork are run only if TW is master (not canary)
+            if not is_canary:
```

Comment:
I think it will read better if you put the "short part" of the if first:
```python
if is_canary:
   self.logger.info("This is canary TW %s running.", self.config.TaskWorker.name)
 else:
    calls to select works and lockwork
```

Reply:
Okki